ui/apps/platform/cypress/integration/risk/risk.test.js

```diff
@@ -22,44 +22,57 @@ describe('Risk page', () => {
-            cy.location('search').should(
-                'eq',
-                '?sort[id]=Deployment%20Risk%20Priority&sort[desc]=false'
-            );
-            // Why no request here?
-            cy.get(priorityColumnHeadingSelector).should('have.class', '-sort-asc');
+            /*
+             * Because risk table displays all deployments on only one page in integration testing environment,
+             * sort descending just reverses the rows so first becomes last, and last becomes first.
+             */
             cy.get(`${RiskPageSelectors.table.dataRows} .rt-td:nth-child(5)`).then(
                 ($priorityCells) => {
-                    const firstCellPriority = Number($priorityCells.eq(0).text());
-                    const lastCellPriority = Number(
-                        $priorityCells.eq($priorityCells.length - 1).text()
+                    const priorityFirst = $priorityCells.eq(0).text();
+                    const priorityLast = $priorityCells.eq($priorityCells.length - 1).text();
+
+                    // Initial table state is not sorted by priority.
+                    cy.get(priorityColumnHeadingSelector)
+                        .should('not.have.class', '-sort-asc')
+                        .should('not.have.class', '-sort-desc');
+
+                    // Sort ascending by priority.
+                    cy.get(priorityColumnHeadingSelector).click();
+                    cy.location('search').should(
+                        'eq',
+                        '?sort[id]=Deployment%20Risk%20Priority&sort[desc]=false'
                     );
-                    expect(lastCellPriority).to.be.at.least(firstCellPriority);
-                }
-            );
+                    // No request because response is sorted ascending.
 
-            // Sort descending by priority.
-            cy.get(priorityColumnHeadingSelector).click();
-            cy.location('search').should(
-                'eq',
-                '?sort[id]=Deployment%20Risk%20Priority&sort[desc]=true'
-            );
-            cy.wait('@deploymentswithprocessinfo'); // assume intercept in visitRiskDeployments
-            cy.get(priorityColumnHeadingSelector).should('have.class', '-sort-desc');
-            cy.get(`${RiskPageSelectors.table.dataRows} .rt-td:nth-child(5)`).then(
-                ($priorityCells) => {
-                    const firstCellPriority = Number($priorityCells.eq(0).text());
-                    const lastCellPriority = Number(
-                        $priorityCells.eq($priorityCells.length - 1).text()
+                    // Sort descending by priority.
+                    cy.get(priorityColumnHeadingSelector).click();
+                    cy.location('search').should(
+                        'eq',
+                        '?sort[id]=Deployment%20Risk%20Priority&sort[desc]=true'
+                    );
+                    cy.wait('@deploymentswithprocessinfo'); // assume intercept in visitRiskDeployments
```

Comment:
Is there still a requirement for the `cy.wait` calls here? It seems that with the more robust selectors you have to grab the `first` and `last` priority table rows Cypress's query retry will make the `wait` redundant.

Reply:
Good point. I replaced with comments whether or not there is a request for the sake of future readers.